packages/react/src/exampleTypes/StepSlider.test.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+
+import '@testing-library/jest-dom'
+import userEvent from '@testing-library/user-event'
+import { act, render, screen } from '@testing-library/react'
+
+import StepSlider from './StepSlider'
+
+import testTree from './stories/testTree'
+import { renderWithProviders } from '../utils/test'
+
+describe('StepSlider', () => {
+  beforeEach(() => {
+    renderWithProviders(<StepSlider tree={testTree()} path={['step']} />)
+  })
+
+  it('renders range slider', async () => {
+    const rangeSlider = await screen.getByTestId('chakra-range-slider-track')
+    const rangeSliderThumbs = await screen.getAllByRole('slider')
+    expect(rangeSlider).toBeDefined()
+    expect(rangeSliderThumbs).toHaveLength(2)
+  })
+
+  it('renders min and max values', async () => {
+    const rangeLabel = await screen.getByText('Range')
+    const rangeValue = await screen.getByText('0 - 1,000')
+    expect(rangeLabel).toBeDefined()
+    expect(rangeValue).toBeDefined()
```

Comment:
That's correct. But my intention here was to be the more declarative/explicit as possible, even though in this use case that means being redundant.

Another approach would be replacing the `getByText` calls with `queryByText`, so it doesn't throw error but instead returns nullish values. 

Your approach is perfectly valid as well, but I would be concerned that this test case:
```
it('renders min and max values', async () => {
    await screen.queryByText('Range')
    await screen.queryByText('0 - 1,000')
  })
```

Is less clear than this:
```
it('renders min and max values', async () => {
    const rangeLabel = await screen.queryByText('Range')
    const rangeValue = await screen.queryByText('0 - 1,000')
    expect(rangeLabel).toBeInTheDocument()
    expect(rangeValue).toBeInTheDocument()
  })
```

Due mainly because of the variable name that points to the semantics of the query.

What do you think about that.

Reply:
That's an excellent point! 😄. Agreed. I'll resolve this thread.